docs/using_an_external_type.md

```diff
@@ -0,0 +1,109 @@
+# Using an external Type
+
+
+# Introduction
+
+There are several different external types that may be referenced when writing a controller.
+* A Custom Resource Definition (CRD) that is defined in the current project `kubebuilder create api`.
+* A Core Kubernetes Resources eg. `kubebuilder create api --group apps --version v1 --kind Deployment`.
+* A CRD that is created and installed in another project.
+* A CR defined via an API Aggregation (AA). Aggregated APIs are subordinate APIServers that sit behind the primary API server, which acts as a proxy.
+
+Currrently Kubebuilder handles the first two, CRDs and Core Resources, seamlessly.  External CRDs and CRs created via Aggregation must be scaffolded manually.
+
+In order to use a Kubernetes Custom Resource that has been defined in another project
+you will need to have several items of information.
+* The Domain of the CR
+* The Group under the Domain 
+* The Go import path of the CR Type definition.
+
+The Domain and Group variables have been discussed in other parts of the documentation.  The import path would be located in the project that installs the CR.
+
+Example API Aggregation directory structure
+```
+github.com
+    ├── example
+        ├── myproject
+            ├── apis
+                ├── mygroup
+                   ├── doc.go
+                   ├── install
+                   │   ├── install.go
+                   ├── v1alpha1
+                   │   ├── doc.go
+                   │   ├── register.go
+                   │   ├── types.go
+                   │   ├── zz_generated.deepcopy.go
+```
+
+In the case above the import path would be `github.com/example/myproject/apis/mygroup/v1alpha1`
+
+### Create a project
+
+```
+kubebuilder init --domain $APIDOMAIN --owner "MyCompany"
+```
+
+### Add a controller
+
+be sure to answer no when it asks if you would like to create an api? [Y/n]
+```
+kubebuilder create api --group $APIGROUP --version $APIVERSION --kind MyKind
+
+```
+
+## Edit the Api files.
+
+### Register your Types
+
+Add the following file to the pkg/apis directory
+
+file: pkg/apis/mytype_type.go
+```
+package apis
+
+import (
+	"github.com/username/myapirepo/apis/mygroup/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"log"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/scheme"
+)
+
+func init() {
+	// schemeGroupVersion is group version used to register these objects
+	schemeGroupVersion := schema.GroupVersion{Group: "mygroup.mydomain.com", Version: "v1alpha1"}
```

Comment:
Where is `mydomain` from? Is there any file defining it in API Aggregation directory?

Reply:
In order to import a CRD defined in other projects, we only need to vendor the CRD type definition project package (so said `import path`) and import the `add functions`. We have two steps to do it:
1) If the external CRD is created by `kubebuilder`, just need to copy over `addtoscheme_XXX.go` file from `pkg/apis` of project package where the external CRD defined. And let the `dep ensure --add XXX(external import path)` to handle the dependencies. 
`make` will update the current CRD.
2) Similarly, copy over `add_XXX.go` from controller package where external CRD controller defined into `./pkg/controller` of current project.

This could be a feature that `kubebuilder` can take over. @droot